### PR TITLE
Add config test for num_cpu vs torque queue options

### LINKED ISSUE #7265: Port error message from torque_driver to openpbs_driver
This check is not present in the OpenPBS driver.

Evaluate, then deprecate or port.

https://github.com/equinor/ert/blob/21f0a89677b45fcfd4367e262b9f14f6d5a09230/src/clib/lib/job_queue/torque_driver.cpp#L560-L567

Requires #7232

### PR DESCRIPTION
**Pull request by berland**
**Issue**
Resolves #7265 

**Approach**
_Short description of the approach_

(Screenshot of new behavior in GUI if applicable)

- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Added appropriate release note label
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/config/queue_config.py [hunk 1]
```diff
@@ -115,24 +119,41 @@ def max_running(self) -> int:
...
 
+def _check_num_cpu_requirement(
+    num_cpu: int,
+    queue_system_options: List[Tuple[str, str]],
+) -> None:
+    torque_options = _option_list_to_dict(queue_system_options)
+    num_nodes = int(torque_options.get("NUM_NODES", [1])[0])
+    num_cpus_per_node = int(torque_options.get("NUM_CPUS_PER_NODE", [1])[0])
+    if num_cpu > num_nodes * num_cpus_per_node:
+        raise ConfigValidationError(
```
Comment (xjules) [inline src/ert/config/queue_config.py:151]: I don't get this tbh. If user specifies `NUM_CPU` and PBS (with `NUM_CPUS_PER_NODE` and `NUM_NODES`) then `PBS` takes precedence or is not?

Comment (berland) [reply src/ert/config/queue_config.py:151]: Setting to equality.

**Comment by xjules** [comment]
There is a test: test_ert_config_throws_on_missing_forward_model_job which needs to be fixed too

**Comment by berland** [comment]
> Member

passes now.

**Pull request merged**